Source/ACE.Server/WorldObjects/SkillAlterationDevice.cs

```diff
@@ -0,0 +1,250 @@
+                            player.Session.Network.EnqueueSend(new GameMessagePrivateUpdateSkill(player, currentSkill.Skill, SkillAdvancementClass.Trained, 0, 0, 0));
+                            player.Session.Network.EnqueueSend(new GameMessagePrivateUpdatePropertyInt(player, PropertyInt.AvailableSkillCredits, player.AvailableSkillCredits ?? 0));
+                            player.Session.Network.EnqueueSend(new GameEventWeenieErrorWithString(player.Session, WeenieErrorWithString.YouHaveSucceededUnspecializing_Skill, currentSkill.Skill.ToSentence()));
+                            player.Session.Network.EnqueueSend(new GameEventUseDone(player.Session, WeenieError.None));
+
+                            //Destroy the gem we used successfully
+                            Destroy();
+                            player.Session.Network.EnqueueSend(new GameEventInventoryRemoveObject(player.Session, this));
+
+                            break;
+                        }
+                    }
+                    else if (currentSkill.AdvancementClass == SkillAdvancementClass.Trained)
+                    {
+                        bool isHeritageSkill = false;
+
+                        //Check to see if we're lowering a heritage skill
+                        if (CheckHeritageSkill(player))
+                        {
+                            isHeritageSkill = true;
+                        }
+
+                        if (player.UntrainSkill(currentSkill.Skill, currentSkillCost.TrainingCost, isHeritageSkill))
+                        {
+                            //Unspecialization was successful, notify the client
+                            if (!isHeritageSkill)
+                            {
+                                player.Session.Network.EnqueueSend(new GameMessagePrivateUpdateSkill(player, currentSkill.Skill, SkillAdvancementClass.Untrained, 0, 0, 0));
+                                player.Session.Network.EnqueueSend(new GameMessagePrivateUpdatePropertyInt(player, PropertyInt.AvailableSkillCredits, player.AvailableSkillCredits ?? 0));
+                                player.Session.Network.EnqueueSend(new GameEventWeenieErrorWithString(player.Session, WeenieErrorWithString.YouHaveSucceededUntraining_Skill, currentSkill.Skill.ToSentence()));
+                                player.Session.Network.EnqueueSend(new GameEventUseDone(player.Session, WeenieError.None));
+                            }
+                            else
+                            {
+                                player.Session.Network.EnqueueSend(new GameEventWeenieErrorWithString(player.Session, WeenieErrorWithString.CannotUntrain_SkillButRecoveredXP, currentSkill.Skill.ToSentence()));
+                                player.Session.Network.EnqueueSend(new GameEventUseDone(player.Session, WeenieError.None));
+                            }
+
+                            //Destroy the gem we used successfully
+                            Destroy();
```

Comment:
Should be using player.TryRemoveItemFromInventoryWithNetworking(this, 1) without the line following the Destroy()

Reply:
Fixed